contracts/BadgerVotingShare.sol

```diff
@@ -189,10 +233,89 @@ contract BadgerVotingShare {
-        int256 numerator = int256(aBADGER.liquidReserves()) + int256(aBADGER.utilizedReserves()) - int256(aBADGER.undistributedLpFees());
-        uint256 exchangeRateCurrent = (uint256(numerator) * 1e18) / aBADGER.totalSupply();
-        
-        return exchangeRateCurrent * aBADGER.balanceOf(_voter) / 1e18;
+        int256 numerator = int256(aBADGER.liquidReserves()) +
+            int256(aBADGER.utilizedReserves()) -
+            int256(aBADGER.undistributedLpFees());
+        uint256 exchangeRateCurrent = (uint256(numerator) * 1e18) /
+            aBADGER.totalSupply();
+
+        return (exchangeRateCurrent * aBADGER.balanceOf(_voter)) / 1e18;
+    }
+
+    /*
+        The voter can have BadgerWbtc in Balancer in 4 configurations:
+         * Balancer BPT in wallet
+         * Balancer BPT in sett
+         * Balancer BPT in balancer gauge
+         * Balancer BPT in aura gauge
+        Setts have an additional PPFS that we need to take into account
+    */
+    function _balancerBadgerWbtcBalanceOf(
+        address _voter
+    ) internal view returns (uint256) {
+        bytes32 poolId = badger_wBTC_balancer.getPoolId();
+        (IERC20[] memory tokens, uint256[] memory balances, ) = balancer_vault
+            .getPoolTokens(poolId);
+        uint256 poolBadgerAmount = 0;
+        for (uint i = 0; i < tokens.length; i++) {
+            if (tokens[i] == badger) {
+                poolBadgerAmount = balances[i];
+                break;
+            }
+        }
+
+        uint256 bptTotalSupply = badger_wBTC_balancer.totalSupply();
+        uint256 voterBalance = badger_wBTC_balancer.balanceOf(_voter);
+        uint256 voterVaultBalance = sett_badger_wBTC_balancer.balanceOf(_voter);
+        uint256 vaultPPFS = sett_badger_wBTC_balancer.getPricePerFullShare();
+
+        uint256 bptVotes = (voterBalance * poolBadgerAmount) /
+            bptTotalSupply /
+            1e18;
+        uint256 bptStakedVotes = (bptStakedBadgerWbtc.balanceOf(_voter) *
+            poolBadgerAmount) / 1e18;
+        uint256 bptAuraVotes = (bptAuraBadgerWbtc.balanceOf(_voter) *
+            poolBadgerAmount) / 1e18;
+        uint256 bptSettVotes = (voterVaultBalance *
+            poolBadgerAmount *
+            vaultPPFS) /
+            bptTotalSupply /
+            1e18;
+        return bptVotes + bptStakedVotes + bptAuraVotes + bptSettVotes;
+    }
+
+    /*
+        The voter can have BadgerWbtc in Balancer in 4 configurations:
+         * Balancer BPT in wallet
+         * Balancer BPT in balancer gauge
+         * Balancer BPT in aura gauge
+        Setts have an additional PPFS that we need to take into account
+    */
+    function _balancerBadgerRethBalanceOf(
+        address _voter
+    ) internal view returns (uint256) {
+        bytes32 poolId = badgerRethBalancer.getPoolId();
+        (IERC20[] memory tokens, uint256[] memory balances, ) = balancer_vault
+            .getPoolTokens(poolId);
+        uint256 poolBadgerAmount = 0;
```

Comment:
here i think is not req to make it equal to zero explicitly

Reply:
a0a309fb333c8cb1e20ea07e5af4151e2be99b40